frontend/public/components/utils/resource-icon.tsx

```diff
@@ -2,23 +2,36 @@ import * as React from 'react';
 import * as classNames from 'classnames';
 import * as _ from 'lodash-es';
 
-import { K8sResourceKindReference } from '../../module/k8s';
+import {
+  K8sGroupVersionKind,
+  K8sResourceKindReference,
+  referenceForGroupVersionKind,
+} from '../../module/k8s';
 import { modelFor } from '../../module/k8s/k8s-models';
 import { kindToAbbr } from '../../module/k8s/get-resources';
 
 const MEMO = {};
 
-export const ResourceIcon: React.SFC<ResourceIconProps> = ({ className, kind }) => {
-  // if no kind, return null so an empty icon isn't rendered
-  if (!kind) {
+export const ResourceIcon: React.SFC<ResourceIconProps> = ({
+  className,
+  groupVersionKind,
+  kind,
+}) => {
+  // if no kind or groupVersionKind, return null so an empty icon isn't rendered
+  if (!kind && !groupVersionKind) {
     return null;
   }
-  const memoKey = className ? `${kind}/${className}` : kind;
+  const kindReference =
+    kind ||
+    referenceForGroupVersionKind(groupVersionKind.group)(groupVersionKind.version)(
+      groupVersionKind.kind,
+    );
+  const memoKey = className ? `${kindReference}/${className}` : kindReference;
   if (MEMO[memoKey]) {
     return MEMO[memoKey];
   }
-  const kindObj = modelFor(kind);
-  const kindStr = _.get(kindObj, 'kind', kind);
+  const kindObj = modelFor(kindReference);
+  const kindStr = _.get(kindObj, 'kind', kindReference);
```

Comment:
nit: can avoid lodash, here and below as well

```suggestion
  const kindStr = kindObj?.kind ?? kindReference;
```

Reply:
I suppose it's a good idea to update the code as we move it... I sorta just left whatever we were moving alone as it works so why critique the code. But I do agree with this change, onwards and upwards.